src/giskardpy/plugin_interrupts.py

```diff
@@ -47,19 +50,64 @@ def round_js(self, js):
         """
         return tuple(round(x.position, self.wiggle_precision1) for x in js.values())
 
-
+    #@profile
     def check_fft(self):
-        traj = self.get_god_map().safe_get_data(identifier.trajectory)
+        fft_duration = 3
+        wiggle_sensitivity = 10
+        min_wiggle_frequency = 3
+
+        trajectory = self.get_god_map().safe_get_data(identifier.trajectory)
+
         sample_period = self.get_god_map().safe_get_data(identifier.sample_period)
-        current_time = self.get_god_map().safe_get_data(identifier.time)
-        last_sec = []
-        keys = traj._points[0.0].keys()
-        for time in range(int(current_time-1/sample_period), current_time):
-            # rounded_time = np.round(time / sample_period) * sample_period
-            point = traj._points[time]
-            positions = [point[key].position for key in keys]
-            last_sec.append(positions)
-        last_sec = np.array(last_sec)
+        num_points = int(fft_duration/sample_period)
+
+        latest_points = trajectory._points[next(reversed(trajectory._points))]
+
+        if len(self.joint_dict.keys()) == 0:
+            for key in latest_points.keys():
+                self.joint_dict[key] = []
+
+        for key in latest_points.keys():
+            self.joint_dict[key].append(latest_points[key].velocity)
+
+        if len(self.joint_dict.values()[0]) < num_points:
+            return Status.RUNNING
+
+        if len(self.joint_dict.values()[0]) > num_points:
+            for val in self.joint_dict.values():
+                del(val[0])
+
+        # remove joints that arent moving
+        joints_filtered = [self.joint_dict[key] for key in self.joint_dict.keys() if np.any(abs(np.array(self.joint_dict[key])) > 0.1)]
```

Comment:
It's probably faster to first turn everything into a np array and then doing the filtering.